### PR TITLE
Feature#21/register users

### PR DESCRIPTION
**Pull request by anisa114**
- Routes for Users, Admins, and Organizations
- Models for Users, Admins, and Organizations

### CODE REVIEW

File: api/routes/admins.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+const Organization = require("../../db/models/Organization");
+
+
+router.get("/", async (req, res, next) => {
+    const admins = await Admin.find();
+    return res.send(admins);
+});
+router.get("/login", async (req, res, next) => {
+    let email = req.body.email;
+    Admin.findOne({ email: email }).then((admin) => {
```
Comment (apjama) [inline api/routes/admins.js:15]: this seems really secure

Comment (HalimoG) [reply api/routes/admins.js:15]: only the best for your users

File: api/routes/admins.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+            return res.send(organization)
+        })
+    })
+});
+
+router.post("/", (req, res, next) => {
+    Organization.findOne({ _id: req.body.organization }).then(async (organization) => {
+        const admin = new Admin({
+            email: req.body.email,
+            password: req.body.password,
```
Comment (apjama) [inline api/routes/admins.js:26]: ideally we wouldn't want to save a password as text. we could get hacked. we would want to use some sort of hash. for now, let's just add a comment saying we'll change this later.

File: api/routes/organizations.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+var express = require('express');
+var router = express.Router();
+
+const Organization = require("../../db/models/Organization");
+
+router.get("/", async (req, res, next) => {
```
Comment (apjama) [inline api/routes/organizations.js:6]: who would need to use this end points? seems like a dangerous endpoint cos it exposes all our clients.

File: api/routes/users.js [hunk 4]
```diff
@@ -1,9 +1,42 @@
...
+const Organization = require("../../db/models/Organization");
+
+router.get("/", async (req, res, next) => {
+  const users = await Users.find();
+  return res.send(users);
 });
 
+router.post("/", async (req, res, next) => {
+  let addressBody = req.body.address;
+  const address = new Address({
```
Comment (apjama) [inline api/routes/users.js:18]: can we wrap somehow in some async await ? basically what if one of the three things fails? we want the user to be able to get some helpful feedback. 

this only works if all of these things work.

File: db/models/Address.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    },
+    city: {
+        type: String,
+        required: true
+    },
+    province: {
+        type: Object,
+        required: true
+    },
+    postalCode: {
```
Comment (apjama) [inline db/models/Address.js:18]: this is not a date object.

File: api/routes/users.js (hunk 4 above)
Comment (anisa114) [reply api/routes/users.js:18]: addressed

File: db/models/Address.js (hunk 5 above)
Comment (anisa114) [reply db/models/Address.js:18]: addressed

File: api/routes/organizations.js (hunk 3 above)
Comment (anisa114) [reply api/routes/organizations.js:6]: test case use only

File: api/routes/admins.js (hunk 2 above)
Comment (anisa114) [reply api/routes/admins.js:26]: addressed

**Review by apjama — APPROVED** [review]

**Pull request merged**